components/user_group_popover/useShouldClose.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {useEffect, useState} from 'react';
+import {useSelector} from 'react-redux';
+
+import {getOpenModalCount} from 'selectors/views/modals';
+
+export default function useShouldClose(): boolean {
+    const [shouldClose, setShouldClose] = useState(false);
+    const [initialOpenModalCount, setInitialOpenModalCount] = useState(0);
+    const openModalCount = useSelector(getOpenModalCount);
+
+    useEffect(() => {
+        setInitialOpenModalCount(openModalCount);
+    }, []);
+
+    useEffect(() => {
+        if (initialOpenModalCount !== openModalCount) {
+            setShouldClose(true);
+        }
+    }, [initialOpenModalCount, openModalCount]);
```

Comment:
I commented out this `useEffect` block and the it doesn't seem to change the behaviour... or did I test it wrong?

Reply:
This is for the group info, after @cyrilzhang-mm suggestion. Maybe you've checked the regular profile one? It shouldn't work without it.

Reply:
Yes you are right. Thank you.